test/cql-pytest/test_permissions.py

```diff
@@ -39,3 +40,46 @@ def test_grant_applicable_data_and_role_permissions(cql, test_keyspace, cassandr
             for permission in all_permissions.difference(set(permissions)):
                 with pytest.raises((InvalidRequest, SyntaxException), match="support.*permissions"):
                     cql.execute(f"GRANT {permission} ON {resource} TO {user}")
+
+
+def eventually_authorized(fun, retries=100):
+    for i in range(retries):
+        try:
+            return fun()
+        except Unauthorized as e:
+            time.sleep(0.1)
+    return fun()
+
+def eventually_unauthorized(fun, retries=100):
+    for i in range(retries):
+        try:
+            fun()
+            time.sleep(0.1)
+        except Unauthorized as e:
+            return
+    try:
+        fun()
+        raise Exception(f"Function {fun} was not refused as unauthorized")
```

Comment:
there's also pytest.fail() that perhaps more clearly indicates that this is a test failure, not a normal exception?

Reply:
missed this comment, I'll update in a few minutes